x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/Security.java

```diff
@@ -786,11 +787,16 @@ private static void addTribeSettings(Settings settings, Settings.Builder setting
             }
 
             // we passed all the checks now we need to copy in all of the x-pack security settings
-            settings.keySet().forEach(k -> {
+            SecureSettings secureSettings = Settings.builder().put(settings).getSecureSettings(); // hack to get at secure settings...
+            Set<String> secureSettingKeys = secureSettings == null ? Collections.emptySet() : secureSettings.getSettingNames();
+            for (String k : settings.keySet()) {
                 if (k.startsWith("xpack.security.")) {
+                    if (secureSettingKeys.contains(k)) {
+                        throw new IllegalArgumentException("Secure setting [" + k + "] cannot be used with tribe client node");
```

Comment:
How about building a set of invalid keys here just to be a little friendlier to a user with multiple secure settings